MicroPython/src/communication/wirerless_connection_controller.py

```diff
@@ -79,28 +76,59 @@ def configure_station(self) -> (bool, str):
 
         self.sta_handler.active(True)
         existing_networks = self.sta_handler.scan()
-        print(existing_networks)
-        print("Connect to wifi: {}, pass: {}".format(self.sta_ssid, self.sta_password))
-        try:
-            self.sta_handler.connect(self.sta_ssid, self.sta_password)
-        except Exception:
-            raise Exception("Failed to connect to access point (wifi ssid='{}')".format(self.sta_ssid))
-
-        number_of_retires = 0
-        while not self.sta_handler.isconnected():
-            logging.info("Reconnection while loop")
-            time.sleep(self.WIFI_CONNECTION_CHECK_SLEEP_TIME_S)
-            number_of_retires += 1
-            if number_of_retires >= self.WIFI_CONNECTION_MAX_NUMBER_OF_RETRIES:
-                logging.info("Too many reconnections")
-                break
 
-        if self.sta_handler.isconnected():
-            self.sta_handler.ifconfig()
-            return True, ""
-        else:
-            self.disconnect_station()
-            raise Exception("Failed to connect to access point v2 (wifi ssid='{}')".format(self.sta_ssid))
+        # Sort detected networks according to signal strength in descending order and filter out the ones to
+        # which credentials were not given
+        existing_networks.sort(reverse=True, key=lambda ap: ap[3])
+        logging.info("Networks detected: {}".format(existing_networks))
+
+        existing_networks_also_on_AP_list_sorted = []
+        for ap in existing_networks:
+            for credential_pair in self.sta_access_points:
+                if ap[0].decode('ascii') == credential_pair[
+                    "ssid"] and ap not in existing_networks_also_on_AP_list_sorted:
```

Comment:
Unpack ap to some meaningful variables because I have no idea what ap[0] is. 
Fix formatting - new line character after "and", not after first square bracket.